imagetagger/imagetagger/images/views.py

```diff
@@ -638,6 +639,48 @@ def dl_script(request):
                             }, content_type='text/plain')
 
 
+def download_imageset_zip(request, image_set_id):
+    image_set = get_object_or_404(ImageSet, id=image_set_id)
+
+    if not image_set.has_perm('read', request.user):
+        return HttpResponseForbidden()
+
+    if image_set.image_count == 0:
+        # It should not be possible to download empty image sets. This
+        # is already blocked in the UI, but it should also be checked
+        # on the server side.
+        messages.warning(request, 'There are no images in the selected image set.')
+        return redirect(reverse('images:view_imageset', args=(image_set_id,)))
+
+    if not image_set.zip_finished:
+        # The zip creation process after an upload is still in progress.
+        # Therefore, we do not want to start a second process operating
+        # on the same file.
+        messages.warning(request, 'The zip file generation is still in progress. Try again in a few minutes.')
+        return redirect(reverse('images:view_imageset', args=(image_set_id,)))
+
+    if not image_set.zip_file_exists():
+        # The zipfile does not yet exist and since its creation might take
+        # some time, the user should receive a feedback.
+        image_set.update_zip()
+        messages.warning(request, 'The zip file generation was initiated. The archive will be ready in a few minutes.')
+        return redirect(reverse('images:view_imageset', args=(image_set_id,)))
+
+    # Even though the zip file should represent the current state of the
+    # directory, update_zip is executed to be double safe. If there are
+    # no changes, it will take less than a second. Threading is disabled
+    # to prevent the download of an unfinished zip archive.
+    image_set.update_zip(threading=False)
+
+    chunk_size = 8192
+    file_path = os.path.join(settings.IMAGE_PATH, image_set.relative_zip_path())
+    response = StreamingHttpResponse(FileWrapper(open(file_path, 'rb'), chunk_size),
+                                     content_type='application/zip')
+    response['Content-Length'] = os.path.getsize(file_path)
+    response['Content-Disposition'] = "attachment; filename={}".format(image_set.name + '.zip')
```

Comment:
Please add support for direct download via nginx, see imagedownload

Reply:
You can't use the name of an image set as file name without filtering it. For example, an imageset is allowed to contain any character; slashes as well as other special characters will cause problems.